promql/engine_test.go

```diff
@@ -2161,26 +2166,47 @@ func NewFakeQueryLogger() *FakeQueryLogger {
 	}
 }
 
-// It implements the promql.QueryLogger interface.
 func (f *FakeQueryLogger) Close() error {
 	f.closed = true
 	return nil
 }
 
-// It implements the promql.QueryLogger interface.
-func (f *FakeQueryLogger) Log(ctx context.Context, level slog.Level, msg string, args ...any) {
+func (f *FakeQueryLogger) Enabled(ctx context.Context, level slog.Level) bool {
+	return true
+}
+
+func (f *FakeQueryLogger) Handle(ctx context.Context, r slog.Record) error {
 	// Test usage only really cares about existence of keyvals passed in
 	// via args, just append in the log message before handling the
 	// provided args and any embedded kvs added via `.With()` on f.attrs.
-	log := append([]any{msg}, args...)
-	log = append(log, f.attrs...)
+	log := append([]any{r.Message}, f.attrs...)
 	f.attrs = f.attrs[:0]
+
+	r.Attrs(func(a slog.Attr) bool {
+		log = append(log, a.Key, a.Value.Any())
```

Comment:
I think this could be simplified by holding `[]slog.Attr` and modifying the one test that looks at them.

Reply:
If it's cool to rework the test itself, then I'd much rather make the test create a `t.TempDir()` and make some proper temp files to use for the query logger testing and do the validation against the files. 

There's a utility function in the json file logger's test suite `getLogLines` that is helpful in this regard. I can't think of a direct/easy way to export a test function offhand, but the function is small and neat enough to port over IMO.

Reply:
No principled objection to using files.